src/settings.c

```diff
@@ -902,26 +1168,15 @@ static int setting_perform_request_reply(settings_t *ctx,
  */
 static int setting_register(settings_t *ctx, setting_data_t *setting_data)
 {
-  /* Build message */
-  char msg[SBP_PAYLOAD_SIZE_MAX];
-  uint8_t msg_len = 0;
+  char msg[SBP_PAYLOAD_SIZE_MAX] = {0};
   uint8_t msg_header_len;
-  int l;
 
-  l = message_header_get(setting_data, &msg[msg_len], sizeof(msg) - msg_len);
-  if (l < 0) {
-    ctx->api_impl.log(log_err, "error building settings message");
-    return -1;
-  }
-  msg_len += l;
-  msg_header_len = msg_len;
+  int msg_len = setting_format_setting(setting_data, msg, sizeof(msg), &msg_header_len);
 
-  l = message_data_get(setting_data, &msg[msg_len], sizeof(msg) - msg_len);
-  if (l < 0) {
-    ctx->api_impl.log(log_err, "error building settings message");
+  if (msg_len < 0) {
+    ctx->api_impl.log(log_err, "%s: Setting format failed", __FUNCTION__);
```

Comment:
In several other occasions we print error logs without `__FUNCTION__`.

Reply:
Maybe a local macro would be useful here where you could add `__FUNCTION__` or `__FILE__` & `__LINE__`.

Reply:
I'll remove the `__FUNCTION__` for now. I'll come up with a unified log scheme in a separate PR.